compiler/rustc_interface/src/util.rs

```diff
@@ -48,12 +48,20 @@ pub fn add_configuration(cfg: &mut Cfg, sess: &mut Session, codegen_backend: &dy
     }
 }
 
-const STACK_SIZE: usize = 8 * 1024 * 1024;
-
-fn get_stack_size() -> Option<usize> {
-    // FIXME: Hacks on hacks. If the env is trying to override the stack size
-    // then *don't* set it explicitly.
-    env::var_os("RUST_MIN_STACK").is_none().then_some(STACK_SIZE)
+pub static STACK_SIZE: OnceLock<usize> = OnceLock::new();
+pub const DEFAULT_STACK_SIZE: usize = 8 * 1024 * 1024;
+
+fn init_stack_size() -> usize {
+    // Obey the environment setting or default
+    *STACK_SIZE.get_or_init(|| {
+        env::var_os("RUST_MIN_STACK")
+            .map(|os_str| os_str.to_string_lossy().into_owned())
+            // ignore if it is set to nothing
+            .filter(|s| s.trim() != "")
+            .map(|s| s.trim().parse::<usize>().unwrap())
```

Comment:
Do we want to `.unwrap()` here? The UX on a bad `RUST_MIN_STACK` wouldn't be great. Given the usage of `.filter()` and `.unwrap_or()` outside of this, does it mean we want to silently ignore bad values (_e.g._, `RUST_MIN_STACK="   "`)?

```suggestion
            .and_then(|s| s.trim().parse::<usize>().ok())
```

or, if we don't want to silently ignore the erroneous value:

```suggestion
            .map(|s| s.trim().parse::<usize>().expect("`RUST_MIN_STACK` env var to be a `usize`))
```

Reply:
> thread 'main' panicked at compiler/rustc_interface/src/util.rs:61:48:
> called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }

Huh, I thought for some reason it'd look better than that.